File: src/codes/no-undo-history/ts/uncode.ts
class TextEditor {
  private content: string = "";

  type(text: string) {
    this.content += text;
  }

  getContent() {
    return this.content;
  }
}

const editor = new TextEditor();
editor.type("Hello ");
editor.type("World");

console.log(editor.getContent()); // Hello World

// 取り消しができない！

File: src/codes/no-undo-history/py/uncode.py
class TextEditor:
    def __init__(self):
        self.content = ""

    def type(self, text: str):
        self.content += text

    def get_content(self) -> str:
        return self.content

# 利用例
editor = TextEditor()
editor.type("Hello ")
editor.type("World")

print(editor.get_content())  # Hello World

# 取り消しができない！
